spec/models/reducers/summary_statistics_reducer_spec.rb

```diff
@@ -0,0 +1,225 @@
+      expect(r4.errors[:operations]).to be_present
+
+      r5 = described_class.new(workflow_id: workflow.id, config: {"operations" => [0]})
+      r5.validate
+      expect(r5.errors[:operations]).to be_present
+
+      r6 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["blah"]})
+      r6.validate
+      expect(r6.errors[:operations]).to be_present
+
+      r7 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["sum"]})
+      r7.validate
+      expect(r7.errors[:operations]).not_to be_present
+
+      r8 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["sum", 0]})
+      r8.validate
+      expect(r8.errors[:operations]).to be_present
+
+      r9 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["sum", "blah"]})
+      r9.validate
+      expect(r9.errors[:operations]).to be_present
+
+      r10 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["sum", "mean"]})
+      r10.validate
+      expect(r10.errors[:operations]).not_to be_present
+
+    end
+
+    it 'reads the config fields correctly' do
+      r1 = described_class.new(workflow_id: workflow.id, config: {"summarize_field" => "simple_field"})
+      r2 = described_class.new(workflow_id: workflow.id, config: {"summarize_field" => "complex.field"})
+
+      expect(r1.send(:summarize_field)).to eq("simple_field")
+      expect(r2.send(:summarize_field)).to eq("complex.field")
+
+      expect(r1.send(:extractor_name)).to be_nil
+      expect(r2.send(:extractor_name)).not_to be_nil
+      expect(r2.send(:extractor_name)).to eq("complex")
+
+      expect(r1.send(:field_name)).to eq("simple_field")
+      expect(r2.send(:field_name)).to eq("field")
+
+      r3 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["sum", "mean"]})
+      expect(r3.send(:operations)).to eq(["sum", "mean"])
+
+      r4 = described_class.new(workflow_id: workflow.id, config: {"operations" => "sum"})
+      expect(r4.send(:operations)).to eq(["sum"])
+
+      r5 = described_class.new(workflow_id: workflow.id, config: {"operations" => ["sum","sum"]})
+      expect(r5.send(:operations)).to eq(["sum"])
+    end
+  end
+
+  describe '#reduction_data_for' do
+    it 'considers only relevant extracts' do
+      r1 = described_class.new(workflow_id: workflow.id, config: {"summarize_field" => "simple_field"})
+      r2 = described_class.new(workflow_id: workflow.id, config: {"summarize_field" => "complex.field"})
+
+      extracts = [
+        create(:extract, extractor_key: "foo"),
+        create(:extract, extractor_key: "complex"),
+        create(:extract, extractor_key: "foo"),
+        create(:extract, extractor_key: "complex"),
+        create(:extract, extractor_key: "complex")
+      ]
+
+      allow(r1).to receive(:extracts).and_return(extracts)
+      allow(r2).to receive(:extracts).and_return(extracts)
+
+      expect(r1.send(:relevant_extracts)).to eq(extracts)
```

Comment:
This will potentially get reworked anyway, but both stubbing things on the object under test and testing private methods are often code smells. Leaving a comment here to clarify further and help come up with solutions if this smell doesn't go away by the next review cycle because of other comments posted.

Reply:
It probably wasn't necessary to test `relevant_extracts` separately from testing `extract_data_for` but I was writing the tests as I built the code out. When that sort of thing happens, is it a better idea to just remove the tests rather than keeping them working?